### PR TITLE
adding weights option to fill_events

### PR DESCRIPTION
**Pull request by MRegeard**
This pull request is related to #4457. 
I added a weights option to `Map.fill_events`. Since `Map.fill_events` uses `Map.fill_by_coord` which has a weights option, the implementation is straight forward.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @MRegeard! I'm not sure we would really like to expose this, where we explicitly refer to "events" (see [my comment ](https://github.com/gammapy/gammapy/issues/4457#issuecomment-1499038177)). What do you think @registerrier ?

File: gammapy/maps/core.py [hunk 1]
```diff
@@ -1088,9 +1088,17 @@ def reproject_to_geom(self, geom, preserve_counts=False, precision_factor=10):
...
-        self.fill_by_coord(events.map_coord(self.geom))
+    def fill_events(self, events, weights=None):
+        """Fill event coordinates (`~gammapy.data.EventList`).
+
+        Parameters
+        ----------
+        events : `~gammapy.data.EventList`
+            Events to be fill in the map.
+        weights : `~numpy.ndarray`
+            Weights vector. Default is weight of one.
```
Comment (adonath) [inline gammapy/maps/core.py:1099]: Add that it should have the same shape as `events` or be broadcastable (not sure what is correct...)

**Review by adonath — COMMENTED** [review]
Thanks @MRegeard! Please add a minimal test to establish the behavior...

**Review by registerrier — COMMENTED** [review]
Thanks @MRegeard . Looks good. Please a a test for this new functionality.

**Review by registerrier — APPROVED** [review]
Thanks @MRegeard . No further comment.

**Pull request merged**